### PR TITLE
[KOGITO-2640] Upgrade tests to godog 0.10.0

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2640

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: test/main_test.go [hunk 1]
```diff
@@ -133,31 +133,35 @@ func configureTestOutput() {
...
+}
+
+func InitializeTestSuite(ctx *godog.TestSuiteContext) {
+	// Nothing done here for now
 }
 
-func FeatureContext(s *godog.Suite) error {
+func InitializeScenario(ctx *godog.ScenarioContext) {
 	// Create kube client
 	if err := framework.InitKubeClient(); err != nil {
```
Comment (sutaakar) [inline test/main_test.go:145]: Would it have sense to move InitKubeClient and CheckSetup to InitializeTestSuite ? IMHO those are steps which need to be done just once.

Comment (radtriste) [reply test/main_test.go:145]: make sense yep. gonna move them

File: test/steps/data.go [hunk 2]
```diff
@@ -39,32 +38,32 @@ type Data struct {
...
+	registerOperatorSteps(ctx, data)
+	registerPrometheusSteps(ctx, data)
+	registerProcessSteps(ctx, data)
+	registerTaskSteps(ctx, data)
+	registerKogitoDeployFilesSteps(ctx, data)
 }
 
 // BeforeScenario configure the data before a scenario is launched
-func (data *Data) BeforeScenario(pickle *messages.Pickle) error {
+func (data *Data) BeforeScenario(pickle *godog.Scenario) error {
```
Comment (sutaakar) [inline test/steps/data.go:66]:
```suggestion
func (data *Data) BeforeScenario(scenario *godog.Scenario) error {
```

Comment (sutaakar) [reply test/steps/data.go:66]: Also to be applied in changes below

Comment (radtriste) [reply test/steps/data.go:66]: yep

**Pull request merged**